hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/MapDeleteMessageTask.java

```diff
@@ -32,6 +33,8 @@
 public class MapDeleteMessageTask
         extends AbstractMapPartitionMessageTask<MapDeleteCodec.RequestParameters> {
 
+    private transient long startTimeNanos;
```

Comment:
do we need to use volatile here? seems like `beforeProcess()` and `beforeResponse()` can be called from different threads.

Reply:
There is a happens-before relation between `beforeProcess` and `beforeResponse`. So all writes in `beforeProcess` is visible to `beforeResponse`. 

Happens before relation is there because they are using the same future underneath. The memory barrier is here the future's response.